test/jdk/java/lang/RuntimeTests/loadLibrary/aix/LoadAIXLibraryFromArchiveObject.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2024, IBM Corporation. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug 8319516
+ * @requires os.family == "aix"
+ * @run  main/othervm -Djava.library.path=/usr/lib LoadLibraryTestAIX
+ */
+
+public class LoadAIXLibraryFromArchiveObject {
+    public static void main(String[] args) throws Exception {
+        String libraryName = "perfstat";
```

Comment:
I suggest to check if `/usr/lib/libperstat.a` is present and `/usr/lib/libperfstat.so` is not present; otherwise, the test should throw RuntimeException.

Reply:
I understand. But isn't the presence of .so also a legit test case ? I mean loadLibrary is not expected to fail if that happens. The failure is expected when .so is not present and .a is also not present . Shouldn't we check for non existence of both ?